rbd-nbd: add --legacy and make --try-netlink as deprecated (#37279)
ceph/ceph: doc/man/8/rbd-nbd.rst

   Override device timeout. Linux kernel will default to a 30 second request timeout.
   Allow the user to optionally specify an alternate timeout.

.. option:: --legacy

>    Use the nbd ioctl interface (old kernels), using the netlink interface as default.

Reviewer: ```suggestion
   Use the NBD ioctl interface (old kernels) instead of the default netlink interface.
```
